tests/test_main.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from os import remove
+from os.path import join
+
+import pytest
+from hapi_schema.db_admin1 import DBAdmin1
+from hapi_schema.db_admin2 import DBAdmin2
+from hapi_schema.db_age_range import DBAgeRange
+from hapi_schema.db_dataset import DBDataset
+from hapi_schema.db_gender import DBGender
+from hapi_schema.db_location import DBLocation
+from hapi_schema.db_operational_presence import DBOperationalPresence
+from hapi_schema.db_org import DBOrg
+from hapi_schema.db_orgtype import DBOrgType
+from hapi_schema.db_population import DBPopulation
+from hapi_schema.db_resource import DBResource
+from hapi_schema.db_sector import DBSector
+from hdx.api.configuration import Configuration
+from hdx.database import Database
+from hdx.scraper.utilities.reader import Read
+from hdx.utilities.dateparse import parse_date
+from hdx.utilities.errors_onexit import ErrorsOnExit
+from hdx.utilities.path import temp_dir
+from hdx.utilities.useragent import UserAgent
+from sqlalchemy import func, select
+
+from hapi.pipelines.app import load_yamls
+from hapi.pipelines.app.pipelines import Pipelines
+
+logger = logging.getLogger(__name__)
+
+
+class TestHAPIPipelines:
+    @pytest.fixture(scope="function")
+    def configuration(self):
+        UserAgent.set_global("test")
+        project_configs = [
+            "core.yaml",
+            "population.yaml",
+            "operational_presence.yaml",
+        ]
+        project_config_dict = load_yamls(project_configs)
+        Configuration._create(
+            hdx_read_only=True,
+            hdx_site="prod",
+            project_config_dict=project_config_dict,
+        )
+        return Configuration.read()
+
+    @pytest.fixture(scope="function")
+    def folder(self):
+        return join("tests", "fixtures")
+
+    def test_pipelines(self, configuration, folder):
+        with ErrorsOnExit() as errors_on_exit:
+            with temp_dir(
+                "TestHAPIPipelines",
+                delete_on_success=True,
+                delete_on_failure=False,
+            ) as temp_folder:
+                dbpath = join(temp_folder, "test_hapi.db")
+                try:
+                    remove(dbpath)
+                except OSError:
+                    pass
+                params = {
+                    "dialect": "sqlite",
+                    "database": dbpath,
+                }
+                logger.info(f"Creating database {dbpath}")
+                with Database(**params) as session:
+                    today = parse_date("2023-10-11")
+                    Read.create_readers(
+                        temp_folder,
```

Comment:
For me it would be helpful (easier to follow / understand) if you passed the parameter names explicitly here

Reply:
Good idea!